src/main/java/org/folio/event/handler/InventoryCreateAsyncRecordHandler.java

```diff
@@ -69,10 +71,10 @@ private Future<String> getTenantId(Map<String, String> headers) {
     return consortiumConfigurationService.getConsortiumConfiguration(headers)
       .map(optionalConsortiumConfiguration -> optionalConsortiumConfiguration
         .map(ConsortiumConfiguration::centralTenantId)
-        .orElse(headersTenantId))
+        .orElseThrow(() -> new IllegalStateException(CONSORTIUM_CONFIG_NOT_FOUND)))
```

Comment:
we need to rewrite logic to avoid building it on throwing/catching exceptions.
See more at chapter: "Use exceptions only for exceptional conditions." of Effective Java
And also having errors in logs can confuse that something went wrong. You can just return from this method centralTenantId and from the upper layer in base class check that if incoming kafka message is not from central tenant, just skip processing, probably even debug level log would be enough.

Reply:
Thanks for suggestion, sounds better to get rid of exception. However I think we should not consider headers' tenantId as to be a failed scenario, as it could be the same as central tenant id. Better to set it null and then to the null-check before processing

Reply:
looks good now